File: src/engine/ChangeLens.Engine/AnalysisRuns/Models/AnalysisGetActiveResult.cs
using System.Text.Json.Serialization;

namespace ChangeLens.Engine.AnalysisRuns.Models;

/// <summary>
///     Represents the tagged active-run lookup union in the engine protocol.
/// </summary>
[JsonPolymorphic(TypeDiscriminatorPropertyName = "state")]
[JsonDerivedType(typeof(NoneAnalysisGetActiveResult), "none")]
[JsonDerivedType(typeof(ActiveAnalysisGetActiveResult), "active")]
internal abstract record AnalysisGetActiveResult;

File: src/desktop/src-tauri/src/analysis/models/analysis_get_active_result.rs
use crate::analysis::models::AnalysisRunProjection;
use serde::{Deserialize, Serialize};

/// Represents the tagged active-run lookup union in the engine protocol.
#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
#[serde(
    tag = "state",
    rename_all = "camelCase",
    rename_all_fields = "camelCase",
    deny_unknown_fields
)]
pub enum AnalysisGetActiveResult {
    None,
    Active { run: Box<AnalysisRunProjection> },
}
